### PR TITLE
fix: Terminator bionic arms errors

### PR DESCRIPTION
**Pull request by EttyKitty**
- Fix a [crash](https://discord.com/channels/714022226810372107/1320107379798577183).
- Refactor and format arm drawing code.
- Cleanup garbage files.
- Account for some edge cases.

### CODE REVIEW

File: scripts/scr_draw_unit_image/scr_draw_unit_image.gml [hunk 1]
```diff
@@ -331,61 +331,62 @@ function scr_draw_unit_image(_background=false){
...
-                    var bionic_spr_index = bionic_arm.variant * 2;
+                var _bionic_arm = get_body_data("bionic", right_left == 1 ? "right_arm" : "left_arm");
+                if (arm_variant[right_left] == 1 && _bionic_spr != undefined && !hide_bionics && _bionic_arm) {
+                    var _bionic_variant = _bionic_arm.variant * 2;
+                    var bionic_spr_index = 0;
                     if (right_left == 2) {
-                        bionic_spr_index += (specialist_colours >= 2) ? 1 : 0;
-                        draw_sprite_flipped(spr_bionics_arm, bionic_spr_index, offset_x, offset_y);
+                        _bionic_variant += (specialist_colours >= 2) ? 1 : 0;
+                        bionic_spr_index = _bionic_variant % sprite_get_number(_bionic_spr) - 1;
```
[the remark this answers is not in the document]
Comment (EttyKitty) [reply scripts/scr_draw_unit_image/scr_draw_unit_image.gml]: are you sure if that's possible? if the sprite exists, it'll have a minimum of 1 subimg.
this means, that `sprite_get_number(_bionic_spr) - 1` can never go below 0.
am I misunderstanding something?

**Comment by EttyKitty** [comment]
@sourcery-ai review

**Comment by OH296** [comment]
[dev-0.10] gml_Script_anon@12198@scr_draw_unit_image_scr_draw_unit_image (line 353)
```log
############################################################################################
Your game just encountered and caught an error!

Date-Time: 24-12-2024 17:52:42
Game Version: dev-0.10
Build Date: 
Commit Hash: 

Details:
ERROR in
action number 1
of  Step Event0
for object obj_controller:


Variable TTRPG_stats.spr_bionics_arm_2(100437, -2147483648) not set before reading it.
 at gml_Script_anon@12198@scr_draw_unit_image_scr_draw_unit_image (line 353) -                     _bionic_options = [spr_bionics_arm, spr_bionics_arm_2];


Stacktrace:
gml_Script_anon@12198@scr_draw_unit_image_scr_draw_unit_image (line 353)
gml_Script_scr_draw_unit_image (line 1235) -                 draw_unit_arms(x_surface_offset, y_surface_offset, armour_type, specialist_colours, hide_bionics, complex_set);
gml_Object_obj_controller_Step_0 (line 348) -             temp[121] = unit.draw_unit_image();
############################################################################################
```
https://github.com/EttyKitty/ChapterMaster/commits/

**Comment by OH296** [comment]
actually ignore this thinks it's just my computer having a stroke

**Comment by OH296** [comment]
ok actually was a real error fixed by #238

**Comment by EttyKitty** [comment]
> ok actually was a real error fixed by #238

thanks. it's a bad habit to ignore yyp in the diff all the time.

**Pull request merged**